src/proto/h1/role.rs

```diff
@@ -1468,6 +1474,11 @@ impl Client {
     }
 }
 
+#[cfg(feature = "client")]
+fn is_method_assumed_empty(method: &Method) -> bool {
+    matches!(method, &Method::GET | &Method::HEAD | &Method::CONNECT)
```

Comment:
I'd think `OPTIONS` and `DELETE` should be included here? Probably also `TRACE`.